src/gateway/GasService.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.21;
+
+import {Auth} from "src/Auth.sol";
+import {IGasService} from "src/interfaces/gateway/IGasService.sol";
+import {MathLib} from "src/libraries/MathLib.sol";
+import {MessagesLib} from "src/libraries/MessagesLib.sol";
+
+contract GasService is IGasService, Auth {
+    using MathLib for uint64;
+    using MathLib for uint256;
+
+    /// @inheritdoc IGasService
+    uint64 public proofCost;
+    /// @inheritdoc IGasService
+    uint64 public messageCost;
+    /// @inheritdoc IGasService
+    uint64 public gasPrice;
+    /// @inheritdoc IGasService
+    uint256 public tokenPrice;
+
+    constructor(uint64 proofCost_, uint64 messageCost_, uint64 gasPrice_, uint256 tokenPrice_) {
```

Comment:
why are the proof and messageCost values `uint64`, they can be `uint256`

Reply:
This is something we agreed with Jeroen.